lib/flight_scheduler/commands/batch.rb

```diff
@@ -35,6 +35,7 @@ def run
         job = JobsRecord.create(arguments: args[1..-1],
                                 script_name: File.basename(script_path),
                                 script: script_body,
+                                array: merged_opts.array,
```

Comment:
This `merged_opts.array` may have unexpected outcomes if both `--array` is specified as a magic comment and CLI option. Currently the options only use a shallow merge so the arrays will not be concatenated.

There are three solutions:
1. Don't worry about it as they are supposed to be overridden,
2. Use a deep merge algorithm,
3. Some how parse the `SBATCH` magic comments with the CLI options and leverage the `slop.array` input,

Reply:
If both a magic comment and a corresponding command-line option are given the magic comment should be overridden entirely by the corresponding command-line option.  I can't think of any exceptions to this rule. See my comment above for an explanation of why it should be this way.

Reply:
:+1: